src/Common/Core/Impl/Collections/EnumerableExtensions.cs

```diff
@@ -113,5 +113,11 @@ public static class EnumerableExtensions {
                 }
             }
         }
+
+        public static void ForEach<T>(this IEnumerable<T> enumerable, Action<T> action) {
```

Comment:
https://blogs.msdn.microsoft.com/ericlippert/2009/05/18/foreach-vs-foreach/
TL;DR: there should be no LINQ methods that cause only side effects